core/chains/evm/txm/txm_test.go

```diff
@@ -0,0 +1,246 @@
+		lggr := logger.Test(t)
+		txStore := storage.NewInMemoryStoreManager(lggr, testutils.FixtureChainID)
+		require.NoError(t, txStore.Add(address))
+		txm := NewTxm(lggr, testutils.FixtureChainID, client, ab, txStore, nil, config, keystore)
+		txm.setNonce(address, 8)
+		metrics, err := NewTxmMetrics(testutils.FixtureChainID)
+		require.NoError(t, err)
+		txm.metrics = metrics
+		IDK := "IDK"
+		txRequest := &types.TxRequest{
+			Data:              []byte{100, 200},
+			IdempotencyKey:    &IDK,
+			ChainID:           testutils.FixtureChainID,
+			FromAddress:       address,
+			ToAddress:         testutils.NewAddress(),
+			SpecifiedGasLimit: 22000,
+		}
+		tx, err := txm.CreateTransaction(tests.Context(t), txRequest)
+		require.NoError(t, err)
+		attempt := &types.Attempt{
+			TxID:     tx.ID,
+			Fee:      gas.EvmFee{GasPrice: assets.NewWeiI(1)},
+			GasLimit: 22000,
+		}
+		ab.On("NewAttempt", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(attempt, nil).Once()
+		client.On("SendTransaction", mock.Anything, mock.Anything, mock.Anything).Return(nil).Once()
+
+		bo, err := txm.broadcastTransaction(ctx, address)
+		require.NoError(t, err)
+		assert.False(t, bo)
+		assert.Equal(t, uint64(9), txm.getNonce(address))
+		tx, err = txStore.FindTxWithIdempotencyKey(tests.Context(t), IDK)
+		require.NoError(t, err)
+		assert.Len(t, tx.Attempts, 1)
+		var zeroTime time.Time
+		assert.Greater(t, *tx.LastBroadcastAt, zeroTime)
+		assert.Greater(t, *tx.Attempts[0].BroadcastAt, zeroTime)
+		assert.Greater(t, *tx.InitialBroadcastAt, zeroTime)
+	})
+}
+
+func TestBackfillTransactions(t *testing.T) {
+	t.Parallel()
+
+	ctx := tests.Context(t)
+	client := mocks.NewClient(t)
+	ab := mocks.NewAttemptBuilder(t)
+	storage := mocks.NewTxStore(t)
+	config := Config{}
+	address := testutils.NewAddress()
+	keystore := mocks.NewKeystore(t)
+
+	t.Run("fails if latest nonce fetching fails", func(t *testing.T) {
+		txm := NewTxm(logger.Test(t), testutils.FixtureChainID, client, ab, storage, nil, config, keystore)
+		client.On("NonceAt", mock.Anything, address, mock.Anything).Return(uint64(0), errors.New("latest nonce fail")).Once()
+		bo, err := txm.backfillTransactions(ctx, address)
+		require.Error(t, err)
+		assert.False(t, bo)
+		require.ErrorContains(t, err, "latest nonce fail")
+	})
+
+	t.Run("fails if MarkConfirmedAndReorgedTransactions fails", func(t *testing.T) {
+		txm := NewTxm(logger.Test(t), testutils.FixtureChainID, client, ab, storage, nil, config, keystore)
+		client.On("NonceAt", mock.Anything, address, mock.Anything).Return(uint64(0), nil).Once()
+		storage.On("MarkConfirmedAndReorgedTransactions", mock.Anything, mock.Anything, address).
+			Return([]*types.Transaction{}, []uint64{}, errors.New("marking transactions confirmed failed")).Once()
+		bo, err := txm.backfillTransactions(ctx, address)
+		require.Error(t, err)
+		assert.False(t, bo)
+		require.ErrorContains(t, err, "marking transactions confirmed failed")
+	})
```

Comment:
Shouldn't we also add some success scenarios for backfill? And maybe also one for nonce gap filling?

Reply:
Added two more cases, one for rebroadcast and one for nonce backfill.